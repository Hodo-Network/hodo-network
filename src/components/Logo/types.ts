export interface Props {
	img: string;
	alt: string;
}