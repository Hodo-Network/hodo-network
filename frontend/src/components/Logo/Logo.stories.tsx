import { Meta, Story } from '@storybook/react';

import { HodoPinFilled } from '../../assets';
import { Logo, LogoProps } from './Logo';

export default {
  title: 'Components/Logo',
  component: Logo,
} as Meta;

const Template: Story<LogoProps> = (args) => <Logo {...args} />;

export const Default = Template.bind({});
Default.args = {
  img: HodoPinFilled,
  alt: 'Logo',
};
