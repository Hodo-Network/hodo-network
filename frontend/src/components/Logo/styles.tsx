import styled from 'styled-components';

export const StyledLogo = styled.img`
  max-height: 96px;
  margin: 0 5px 0 0;
`;
