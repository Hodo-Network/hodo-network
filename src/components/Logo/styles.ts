import styled from 'styled-components';

const StyledLogo = styled.img`
	max-height: 96px;
	margin: 0 5px 0 0;
`;

export { StyledLogo };
