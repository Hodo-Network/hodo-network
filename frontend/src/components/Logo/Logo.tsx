import { StyledLogo } from './styles';

export interface LogoProps {
  img: string;
  alt: string;
}

export const Logo = ({ img, alt }: LogoProps) => {
  return <StyledLogo src={img} alt={alt} className='h-6 w-auto' />;
};
