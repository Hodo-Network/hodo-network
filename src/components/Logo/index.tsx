import { FC } from "react";
import { StyledLogo } from "./styles";

const Logo: FC<{ img: string; alt: string }> = ({ img, alt }) => (
  <StyledLogo src={img} alt={alt} className='h-6 w-auto' />
);

export default Logo;
