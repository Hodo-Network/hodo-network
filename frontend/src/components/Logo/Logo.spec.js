import ReactDOM from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Logo from "./index";

afterEach(cleanup);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Logo />, div);
});

test("matches snapshot", () => {
  const tree = renderer.create(<Logo />).toJSON();
  expect(tree).toMatchSnapshot();
});
