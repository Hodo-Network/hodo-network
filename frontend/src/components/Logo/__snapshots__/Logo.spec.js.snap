// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<img
  className="sc-bdfBwQ bIsOIA h-6 w-auto"
/>
`;
